enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java

```diff
@@ -140,6 +140,17 @@ public AbstractNavitiaProvider(final NetworkId network, final String authorizati
 
     protected abstract String region();
 
+    private boolean preferNonNavitiaLineStyles = false;
+
+    /**
+     * Navitia is capable of returning "display_informations" which includes style related information. Depending on the
+     * coverage though, this may not be complete. Prefering non-Navitia line styles means we first delegate to
+     * {@link #lineStyle(String, Product, String)} before then falling back to {@link #getLineStyle(Product, String, String)}.
+     */
```

Comment:
One problem is that sometimes `display_informations` is present, but not sufficient. For example, the state of Tasmania shows a green background for each bus, but the actual bus routes are coloured (I didn't override them in this PR, but will in the future).

However, I think perhaps @schildbach's suggestion of passing the network through to `getLineStyle()` would be nice. The thing is there is already a very sufficient implementation of `lineStyle()` in `AbstractNetworkProvider`, the one which works in conjunction with `setStyles()`. Over this weekend, I'll remove the hack I put in the Navitia provider, and instead pass through `network`. Then, there will be enough information to delegate to `lineStyles()` and all should be well. Thanks for the feedback.